crates/typst/src/eval/plugin.rs

```diff
@@ -0,0 +1,204 @@
+use crate::diag::SourceResult;
+use crate::util::Bytes;
+use ecow::EcoString;
+use std::sync::Mutex;
+use typst::diag::At;
+use wasmi::{AsContext, Caller, Engine, Func as Function, Linker, Module, Value};
+
+#[derive(Debug)]
+pub struct Plugin(Mutex<PluginInstance>, Bytes);
```

Comment:
It would be great to have at least a short doc comment on all public types and methods.